### PR TITLE
amp-sidebar, expose open and close events

### PR DESCRIPTION
**Pull request by aghassemi**
This PR exposes `sidebarOpen` and `sidebarClose` events.

These events fire after sidebar has finished transition and are needed for cases such as focusing on an element in the sidebar (as browsers refuse to focus on invisible items) or running an animation in the sidebar (which only makes sense to start when sidebar is fully visible).

Fixes https://github.com/ampproject/amphtml/issues/12240

### CODE REVIEW

**Comment by aghassemi** [comment]
@camelburrito mmm I noticed we have sidebar 1.0, I thought toobar was added to 0.1, is sidebar 1.0 still used?

File: spec/amp-actions-and-events.md [hunk 1]
```diff
@@ -182,20 +182,37 @@ event.index</pre></td>
   </tr>
 </table>
 
-### amp-selector
```
Comment (camelburrito) [inline spec/amp-actions-and-events.md]: is deleting amp-selector intentional?

File: spec/amp-actions-and-events.md [hunk 2]
```diff
@@ -182,20 +182,37 @@ event.index</pre></td>
...
+    <td>None</td>
+  </tr>
+  <tr>
+    <td><code>sidebarClose</code></td>
+    <td>Fired when sidebar is fully closed after transition has ended.</td>
+    <td>None</td>
+  </tr>
+</table>
+
+### amp-carousel[type="slides"]
```
Comment (camelburrito) [inline spec/amp-actions-and-events.md]: This should already exist ?

Comment (aghassemi) [reply spec/amp-actions-and-events.md]: It's GitHub being bad at showing diffs. Nothing is deleted, just added sidebar above selector.

Comment (aghassemi) [reply spec/amp-actions-and-events.md]: Just GitHub being bad at showing diff. Nothing is deleted, just added sidebar above selector, diff shows up this way.

**Comment by cathyxz** [comment]
Based on this, do we still want it to autoFocus on the opening element after close? Per https://github.com/ampproject/amphtml/issues/12479

**Comment by aghassemi** [comment]
@cathyxz, yes. This allows the page author to implement that feature but honestly most won't do it. We should still do #12479 so all AMP pages would benefit.

**Pull request merged**